nupic/embodied/envs/wrappers.py

```diff
@@ -0,0 +1,539 @@
+            self._env.observation_space = gym.spaces.Box(
+                low=0, high=255.0, shape=(180, 180, 3), dtype="float32"
+            )
+        else:
+            self._env.observation_space = self._env.observation_space["retina"]
+
+    def __getattr__(self, name):
+        return getattr(self._env, name)
+
+    def step(self, action):
+        action = {
+            "joint_command": action,
+            "render": True,
+        }
+        observation, reward, done, info = self._env.step_joints(action)
+        if self.crop_obs:
+            observation = observation["retina"][0:180, 70:250, :]
+        else:
+            observation = observation["retina"]
+        return observation, reward, done, info
+
+    def reset(self):
+        observation = self._env.reset()
+        if self.crop_obs:
+            observation = observation["retina"][0:180, 70:250, :]
+        else:
+            observation = observation["retina"]
+        return observation
+
+
+class CartesianControlDiscrete(object):
+    """
+    Wrapper for the REALRobotEnv (based on MJCFBaseBulletEnv).
+    Changes action space to discrete using cartesian control.
+    Actions: (forward, backward, left, right, up, down, grip, release)
+    """
+
+    def __init__(
+        self, env, crop_obs=False, repeat=1, touch_reward=False, random_force=False
+    ):
+        self._env = env
+        self.crop_obs = crop_obs
+        self.increment_size = 0.5
+        self.repeat = repeat
+        self.touch_reward = touch_reward
+        self.random_force = random_force
+        self.act_dict = {
+            0: "forward",
+            1: "backward",
+            2: "left",
+            3: "right",
+            4: "up",
+            5: "down",
+            6: "grip",
+            7: "release",
+        }
+
+        self._env.action_space = gym.spaces.Discrete(8)
+
+        if self.crop_obs:
+            self._env.observation_space = gym.spaces.Box(
+                low=0, high=255.0, shape=(180, 180, 3), dtype="float32"
+            )
+        else:
+            self._env.observation_space = self._env.observation_space["retina"]
+
+    def __getattr__(self, name):
+        return getattr(self._env, name)
+
+    def step(self, action):
+        cartesian = np.ones(7)
+        gripper = np.zeros(2)
+        loc = self._env.robot.parts["lbr_iiwa_link_7"].get_position()
+        # print('location: '+str(loc))
+        grip = np.array(
+            [
+                self._env.robot.jdict["base_to_finger00_joint"].get_position(),
+                self._env.robot.jdict["finger10_to_finger11_joint"].get_position(),
+            ]
+        )
+        # Currently orientation always makes gripper point down -> could be
+        # changed later and added into action space. (TODO)
+        desiredOrientation = pybullet.getQuaternionFromEuler([0, 3.14, -1.57])
+        cartesian[:3] = loc
+        cartesian[3:] = desiredOrientation
+
+        if action == 0:  # forward
```

Comment:
For readability, what about mapping action to (cartesian_pos, increment_sign) separately from the assignment operation then doing
```
cartesian_pos, increment_sign = action_to_movement[action]
if action in range(5):
    cartesian[cartesian_pos] += self.increment_size * sign
else:
    gripper = grip + self.increment_size * sign
```